### ISSUE TITLE
[Java] C2D Message status is set to Success before a device receives and complete the message

### ISSUE THREAD

**Issue by dluc** [Azure/azure-iot-sdks#993]
When sending a C2D message, and checking for its status, the message status is set to "success" even if the device has not received the message yet.
If the device receives the message and replies with COMPLETE, the status remains "success".

Code: https://gist.github.com/dluc/e24a4eca2faa50e0fc3f66fbf91776d4
The code sends 2 messages. Only the second is received by the device. Both messages have the same feedback status: "success".

Example output:

```
[00:00:31.405] Creating sender client
[00:00:31.621] Open sender connection
[00:00:31.623] Sending message, Content: test, ID: 03bc3c04-6357-48a3-b498-130fe45af73b
[00:00:33.233] Creating feedback receiver client
[00:00:33.235] Open feedback receiver connection
[00:00:33.236] Checking the status of the message...
[00:01:41.528] Retrieving feedback batch...
[00:01:42.521] Found feedback batch, size 1
[00:01:42.521] Batch record 03bc3c04-6357-48a3-b498-130fe45af73b
[00:01:42.522] Message 03bc3c04-6357-48a3-b498-130fe45af73b feedback found: description = Success; status code = success
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
[00:01:42.522] Close feedback receiver connection

[00:01:42.528] Creating sender client
[00:01:42.528] Creating message receiver client
[00:01:42.529] Open sender connection
[00:01:42.530] Sending message, Content: test, ID: 95501c27-d236-43ff-ab2d-8b6aacbccd88
[00:01:42.535] Opening message receiver connection
[00:01:43.487] Message received: Content: test, ID: 95501c27-d236-43ff-ab2d-8b6aacbccd88
[00:01:43.487] Changing the status of the message to COMPLETE 
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
[00:01:48.522] Creating feedback receiver client
[00:01:48.523] Open feedback receiver connection
[00:01:48.523] Checking the status of the message...
[00:01:48.523] Retrieving feedback batch...
[00:01:56.598] Found feedback batch, size 1
[00:01:56.598] Batch record 95501c27-d236-43ff-ab2d-8b6aacbccd88
[00:01:56.598] Message 95501c27-d236-43ff-ab2d-8b6aacbccd88 feedback found: description = Success; status code = success 
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
[00:01:56.599] Close feedback receiver connection
[00:01:56.599] Closing message receiver
[00:01:56.611] Done
```

**Comment by zolvarga** [comment]
Hi Devis,

Correct me if I am wrong but the code what is setting the message status is in our samples.
For the sample purpose I think tat is OK.

Thanks,
Zoltan

**Comment by dluc** [comment]
Hi Zoltan,
sorry I'm not understanding your comment. The problem I'm encountering is that a message sent from an application has the status "success" even if the device has not received it yet. 

The sample code shows that two messages (say A and B), one not received yet (A) and one successfully received (B), have the same status. When an application checks for feedback status, how can it distinguish whether A has been received or not? Given that both have a status of "success" an application would assume that both have been received by the device, right ?

Thanks
Devis

**Comment by dluc** [comment]
After creating a new IoT hub and running the sample code against it, I managed to isolate the root cause. I am generating D2C traffic with a simulator based on [this](https://github.com/Azure/azure-iot-sdk-node/blob/master/device/samples/simple_sample_device.js) code, and after reviewing it I noticed that it was not just sending D2C but also receiving and completing C2D messages, leading to the "success" status describe above. Sorry about the oversight.

Thanks
Devis
